src/utils/requests/apolloRequests.js

```diff
@@ -12,12 +14,34 @@ const client = new ApolloClient({
   }
 });
 
+export function dispatchError(error) {
+  const { message } = error.graphQLErrors[0];
+  store.dispatch(setError(message));
+  store.dispatch(setFetching(false));
+}
+
 export async function apolloQuery(query) {
-  const result = await client.query({ query });
-  return result.data;
+  store.dispatch(setFetching(true));
+  store.dispatch(setError(''));
+  let result;
+  try {
+    result = await client.query({ query });
+    if (result) store.dispatch(setFetching(false));
+  } catch (error) {
+    dispatchError(error);
+  }
+  return result ? result.data : false;
 }
 
 export async function apolloMutate(mutation, variables) {
-  const result = await client.mutate({ mutation, variables });
-  return result.data;
+  store.dispatch(setFetching(true));
+  store.dispatch(setError(''));
+  let result;
+  try {
+    result = await client.mutate({ mutation, variables });
+    if (result) store.dispatch(setFetching(false));
+  } catch (error) {
+    dispatchError(error);
+  }
+  return result ? result.data : false;
```

Comment:
Why do we need to return false if the result is falsey?

Reply:
I believe a ternary must have a return for the "else" part of the statement.

Reply:
Linter was complaining about the return not being in the end of the function :/ so I just set it up to return false if there is no result at the end lol... Plus when you are querying, this is good to just check if there was a result, then just do other things if that was the case